llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -1859,6 +1859,13 @@ class TargetLoweringBase {
     return false;
   }
 
+  /// Whether the atomic operation \p AI with type \p ValueTy and alignment
+  /// \p Alignment via \p PointerTy is natively supported or requires an
+  /// __atomic_* libcall.
+  virtual bool supportsAtomicOperation(const DataLayout &DL,
+                                       const Instruction *AI, Type *ValueTy,
```

Comment:
I could just pass `AI` here, but then I need to add dyn_cast's inside the implementation even though the callers already have that information.

Reply:
This seems fine, it's a common pattern for that reason.